pyirf/interpolation.py

```diff
@@ -2,22 +2,239 @@
+
+    axis: int
+    Axis along which the pdfs used for interpolation are located
+
+    normalize: string or None
+    Mode of normalisation to account for the approximative nature of the interpolation. "sum" normalizes
+    the interpolated histogram to a sum of 1, "weighted_sum" to an integral of 1. None does not apply any
+    normalization.
+
+    quantile_resolution: float
+    Interpolated quantile spacing, defaults to 1/1000
+
+    Returns
+    -------
+    f_new: numpy.ndarray, shape=(...,M,...)
+    Interpolated histograms
+
+    References
+    ----------
+    .. [1] B. E. Hollister and A. T. Pang (2013). Interpolation of Non-Gaussian Probability Distributions
+           for Ensemble Visualization
+           https://engineering.ucsc.edu/sites/default/files/technical-reports/UCSC-SOE-13-13.pdf
+    """
+    # determine quantiles step
+    percentages = np.arange(0, 1 + 0.5 * quantile_resolution, quantile_resolution)
+    mids = bin_center(edges)
+
+    quantiles = np.apply_along_axis(numerical_quantile, axis, hists, mids, percentages)
+
+    # interpolate quantiles step
+    # First: compute alpha from eq. (6), the euclidean norm between the two grid points has to be one,
+    # so normalize to ||m1-m0|| first
+    dist = np.linalg.norm(m[1] - m[0])
+    g = m / dist
+    p = m_prime / dist
+    alpha = np.linalg.norm(p - g[0])
+
+    # Second: Interpolate quantiles as in eq. (10)
+    q_bar = (1 - alpha) * quantiles[0] + alpha * quantiles[1]
+
+    # evaluate interpolant PDF values step
+    # The original PDF (only given as histogram) has to be re-evaluated at the quantiles determined above
+    binnr = np.digitize(quantiles, edges)
+    helper = np.concatenate((hists, binnr), axis=axis)
+    V = np.apply_along_axis(lookup, axis, helper, hists.shape[axis])
+
+    # Compute the interpolated histogram at positions q_bar as in eq. (12), set V_bar to
+    # zero when both template PDFs are zero
+    # V_bar = V[0] * V[1] / ((1 - alpha) * V[1] + alpha * V[0])
+    a = V[0] * V[1]
+    b = (1 - alpha) * V[1] + alpha * V[0]
+    V_bar = np.divide(a, b, out=np.zeros_like(a), where=b != 0)
+
+    # Create temporary axis to imitate the former shape, as one dimension was lost through the interpolation and
+    # therefore axis might not longer be correct
+    q_bar = q_bar[np.newaxis, :]
+    V_bar = V_bar[np.newaxis, :]
+
+    # Shift interpolated pdf back into the original histogram bins as V_bar is by construction given at
+    # positions q_bar.
+    width = np.diff(edges)
+    helper = np.concatenate((q_bar, V_bar), axis=axis)
+    interpolated_histogram = np.apply_along_axis(rebin, axis, helper, mids, width)
+
+    # Re-Normalize, as the normalisation is lost due to approximate nature of this method
+    # Set norm to nan for empty histograms to avoid division through 0
+    if normalize == "sum":
+        norm = np.sum(interpolated_histogram, axis=axis)
+        norm = np.repeat(
```

Comment:
I think there is now need to repeat the values here, you can just rely on numpy broadcasting.